comp/core/tagger/types/entity_id.go

```diff
@@ -0,0 +1,148 @@
+	}
+
+	parts := strings.Split(string(de), "://")
+	if len(parts) != 2 {
+		return ""
+	}
+
+	return EntityIDPrefix(parts[0])
+}
+
+// String implements EntityID#String
+func (de defaultEntityID) String() string {
+	return string(de)
+}
+
+func newDefaultEntityID(id string) EntityID {
+	return defaultEntityID(id)
+}
+
+// compositeEntityID implements EntityID as a struct of prefix and id
+type compositeEntityID struct {
+	Prefix EntityIDPrefix
+	ID     string
+}
+
+// GetPrefix implements EntityID#GetPrefix
+func (eid compositeEntityID) GetPrefix() EntityIDPrefix {
+	return eid.Prefix
+}
+
+// GetID implements EntityID#GetID
+func (eid compositeEntityID) GetID() string {
+	return eid.ID
+}
+
+// String implements EntityID#String
+func (eid compositeEntityID) String() string {
+	return eid.Prefix.ToUID(eid.ID)
+}
+
+// newcompositeEntityID returns a new EntityID based on a prefix and an id
+func newCompositeEntityID(prefix EntityIDPrefix, id string) EntityID {
+	return compositeEntityID{
+		Prefix: prefix,
+		ID:     id,
+	}
+}
+
+// NewEntityID builds and returns an EntityID object based on plain string uid
+// Currently, it defaults to the default implementation of EntityID as a plain string
+func NewEntityID(prefix EntityIDPrefix, id string) EntityID {
+	if config.Datadog().GetBool("tagger.tagstore_use_composite_entity_id") {
+		return newCompositeEntityID(prefix, id)
+	}
+	return newDefaultEntityID(fmt.Sprintf("%s://%s", prefix, id))
+}
+
+// NewEntityIDFromString constructs EntityID from a plain string id
+func NewEntityIDFromString(plainStringID string) (EntityID, error) {
+	if config.Datadog().GetBool("tagger.tagstore_use_composite_entity_id") {
+		if !strings.Contains(plainStringID, "://") {
+			return nil, fmt.Errorf("unsupported tagger entity id format %q, correct format is `{prefix}://{id}`", plainStringID)
+		}
+		parts := strings.Split(plainStringID, "://")
+		return newCompositeEntityID(EntityIDPrefix(parts[0]), parts[1]), nil
+	}
+	return newDefaultEntityID(plainStringID), nil
+}
+
+const (
+	// ContainerID is the prefix `container_id`
+	ContainerID EntityIDPrefix = "container_id"
+	// ContainerImageMetadata is the prefix `container_image_metadata`
+	ContainerImageMetadata EntityIDPrefix = "container_image_metadata"
+	// ECSTask is the prefix `ecs_task`
+	ECSTask EntityIDPrefix = "ecs_task"
+	// Host is the prefix `host`
+	Host EntityIDPrefix = "host"
+	// KubernetesDeployment is the prefix `deployment`
+	KubernetesDeployment EntityIDPrefix = "deployment"
+	// KubernetesMetadata is the prefix `kubernetes_metadata`
+	KubernetesMetadata EntityIDPrefix = "kubernetes_metadata"
+	// KubernetesPodUID is the prefix `kubernetes_pod_uid`
+	KubernetesPodUID EntityIDPrefix = "kubernetes_pod_uid"
+	// Process is the prefix `process`
+	Process EntityIDPrefix = "process"
+)
+
+// AllPrefixesSet returns a set of all possible entity id prefixes that can be used in the tagger
+func AllPrefixesSet() map[EntityIDPrefix]struct{} {
```

Comment:
I don't see this function being used? Is that intended?

Reply:
This function will be used in a subsequent PR to add filters to the subscription logic in the tagger.

I didn't want to add everything in one PR, the PR is already huge 😄

Reply:
in that case I think is best to remove this function from this PR and added in the next one 😉